config/initializers/014-track-setting-changes.rb

```diff
@@ -29,6 +29,21 @@
     end
   end
 
+  # Set bootstrap min users for private sites to a lower default
+  PRIVATE_BOOTSTRAP_MODE_MIN_USERS ||= 10.freeze
+  if name == :login_required && new_value == true && SiteSetting.bootstrap_mode_enabled == true
+    if SiteSetting.bootstrap_mode_min_users == SiteSetting.defaults.get(:bootstrap_mode_min_users)
+      SiteSetting.bootstrap_mode_min_users = PRIVATE_BOOTSTRAP_MODE_MIN_USERS
+    end
+  end
+
+  # Set bootstrap min users for public sites back to the default
+  if name == :login_required && new_value == false && SiteSetting.bootstrap_mode_enabled == true
+    if SiteSetting.bootstrap_mode_min_users == PRIVATE_BOOTSTRAP_MODE_MIN_USERS
+      SiteSetting.bootstrap_mode_min_users = SiteSetting.defaults.get(:bootstrap_mode_min_users)
+    end
+  end
```

Comment:
Shaving a bit of repetition:

```suggestion
  if name == :login_required && SiteSetting.bootstrap_mode_enabled == true
    if new_value == true && SiteSetting.bootstrap_mode_min_users == SiteSetting.defaults.get(:bootstrap_mode_min_users)
      SiteSetting.bootstrap_mode_min_users = PRIVATE_BOOTSTRAP_MODE_MIN_USERS
    end

    # Set bootstrap min users for public sites back to the default
    if new_value == false && SiteSetting.bootstrap_mode_min_users == PRIVATE_BOOTSTRAP_MODE_MIN_USERS
      SiteSetting.bootstrap_mode_min_users = SiteSetting.defaults.get(:bootstrap_mode_min_users)
    end
  end
```